# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/PContent.ipdl
# Commit: bda570180adf
# Full Hash: bda570180adfcc20891d5cfb47406756fe8903ad
# Author: Kagami Sascha Rosylight <saschanaz@outlook.com>
# Date: 2024-11-12 09:32:50
# Regressor Bug: 1928702
# File Overlap Count: 1
# Description:
#   Bug 1928702 - Part 11: Migrate ShowPersistentNotification to use IPC r=asuth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D227696
# ==============================================================================

diff -r c942992fa981 -r bda570180adf dom/ipc/PContent.ipdl
--- a/dom/ipc/PContent.ipdl	Mon Nov 11 20:34:15 2024 +0000
+++ b/dom/ipc/PContent.ipdl	Mon Nov 11 20:34:15 2024 +0000
@@ -412,24 +412,6 @@
     nsCString exceptionHostList;
 };
 
-// Transport structure for Notifications API notifications
-// (https://developer.mozilla.org/en-US/docs/Web/API/notification) instances
-// used exclusively by the NotificationEvent PContent method.
-struct NotificationEventData
-{
-    nsCString originSuffix;
-    nsCString scope;
-    nsString ID;
-    nsString title;
-    nsString dir;
-    nsString lang;
-    nsString body;
-    nsString tag;
-    nsString icon;
-    nsString data;
-    nsString behavior;
-};
-
 struct PostMessageData
 {
     MaybeDiscardedBrowsingContext source;
@@ -1167,22 +1149,6 @@
 
     async AddSecurityState(MaybeDiscardedWindowContext aContext, uint32_t aStateFlags);
 
-    // Request that the ServiceWorkerManager in the parent process create a
-    // notification "click" or "close" event and dispatch it on the relevant
-    // ServiceWorker. This needs to happen because when a notification is
-    // created it is tied to a specific content process and when the user clicks
-    // on the notification, it will be that content process that is notified.
-    // However, even if the ServiceWorker lives in that process (it may no
-    // longer be in that process, or may have never lived there), the right/only
-    // way to talk through the ServiceWorker is through the parent.
-    //
-    // This happens on PContent because the ServiceWorkerManager lives on the
-    // main thread and bouncing this off of PBackground would be silly and
-    // complex. In the long run, the notification implementation will be
-    // overhauled to directly process the notification click/close and directly
-    // translate that to a ServiceWorker event.
-    async NotificationEvent(nsString type, NotificationEventData data);
-
     // Creates a helper for forwarding data from an nsExternalAppHandler
     // running in the content process, to one running in the parent
     // process.